ooniprobe/Model/Database/Measurement.mm

```diff
@@ -96,4 +96,24 @@ -(void)deleteObject{
     [self remove];
 }
 
+- (void)getExplorerUrl:(void (^)(NSString *))measurement_url {
+    NSString *path = [NSString stringWithFormat:@"https://api.ooni.io/api/v1/measurements?report_id=%@&input=%@", self.report_id, self.url_id.url];
+    NSURL *url = [NSURL URLWithString:path];
+    NSURLSessionDataTask *downloadTask = [[NSURLSession sharedSession] dataTaskWithURL:url completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
+        if (!error) {
+            NSDictionary *dic = [NSJSONSerialization JSONObjectWithData:data options:0 error:&error];
+            NSArray *resultsArray = [dic objectForKey:@"results"];
+            if ([resultsArray count] == 0)
+                measurement_url(nil);
+            measurement_url([[resultsArray objectAtIndex:0] objectForKey:@"measurement_url"]);
```

Comment:
The above code seem wrong to me. In the case in which `resultsArray` is empty you end up calling `measurement_url` twice. The  second time, you're going out of bounds.

Reply:
measurement_url is a completion function and can be called only once if I read the documentation correctly. But I can refactor it to call it once

Reply:
> [...] and can be called only once if I read the documentation correctly. But I can refactor it to call it once

Yes, this is a common pattern for callback (or completion) functions. In practice this means that adding a return inside the `if` branch is the expected behaviour.